http/cookie.ts

```diff
@@ -1,10 +1,97 @@
 // Copyright 2018-2019 the Deno authors. All rights reserved. MIT license.
-import { ServerRequest } from "./server.ts";
+import { ServerRequest, Response } from "./server.ts";
+import { assert } from "../testing/asserts.ts";
+import { pad } from "../strings/pad.ts";
 
 export interface Cookie {
   [key: string]: string;
 }
 
+export interface CookieValue {
+  name: string;
+  value: string;
+}
+
+export interface CookieOptions {
+  Expires?: Date;
+  MaxAge?: number;
+  Domain?: string;
+  Path?: string;
+  Secure?: boolean;
+  HttpOnly?: boolean;
+  SameSite?: SameSite;
+}
+
+export type SameSite = "Strict" | "Lax";
+
+function cookieStringFormat(cookie: CookieValue, opt: CookieOptions): string {
+  function dtPad(v: string, lPad: number = 2): string {
+    return pad(v, lPad, { char: "0" });
+  }
+
+  const out: string[] = [];
+  out.push(`${cookie.name}=${cookie.value}`);
+
+  // Fallback for invalid Set-Cookie
+  // ref: https://tools.ietf.org/html/draft-ietf-httpbis-cookie-prefixes-00#section-3.1
+  if (cookie.name.startsWith("__Secure")) {
+    opt.Secure = true;
+  }
+  if (cookie.name.startsWith("__Host")) {
+    opt.Path = "/";
+    opt.Secure = true;
+    delete opt.Domain;
+  }
+
+  if (opt.Secure) {
+    out.push("Secure");
+  }
+  if (opt.HttpOnly) {
+    out.push("HttpOnly");
+  }
+  if (Number.isInteger(opt.MaxAge)) {
+    assert(opt.MaxAge > 0, "Max-Age must be an integer superior to 0");
+    out.push(`Max-Age=${opt.MaxAge}`);
+  }
+  if (opt.Domain) {
+    out.push(`Domain=${opt.Domain}`);
+  }
+  if (opt.SameSite) {
+    out.push(`SameSite=${opt.SameSite}`);
+  }
+  if (opt.Path) {
+    out.push(`Path=${opt.Path}`);
+  }
+  if (opt.Expires) {
+    let dateString = "";
+    let d = dtPad(opt.Expires.getUTCDate().toString());
+    const h = dtPad(opt.Expires.getUTCHours().toString());
+    const min = dtPad(opt.Expires.getUTCMinutes().toString());
+    const s = dtPad(opt.Expires.getUTCSeconds().toString());
+    const y = opt.Expires.getUTCFullYear();
+    // See Date format: https://tools.ietf.org/html/rfc7231#section-7.1.1.1
+    const days = ["Sun", "Mon", "Tue", "Wed", "Thus", "Fri", "Sat"];
+    const months = [
+      "Jan",
+      "Feb",
+      "Mar",
+      "May",
+      "Jun",
+      "Jul",
+      "Aug",
+      "Sep",
+      "Oct",
+      "Nov",
+      "Dec"
+    ];
+    dateString += `${days[opt.Expires.getDay()]}, ${d} ${
+      months[opt.Expires.getUTCMonth()]
+    } ${y} ${h}:${min}:${s} GMT`;
+    out.push(`Expires=${dateString}`);
```

Comment:
Could you break the date formatting out into a standalone function with tests of its own?